consensus/src/chained_bft/block_storage/block_store.rs

```diff
@@ -234,7 +234,9 @@ impl<T: Payload> BlockStore<T> {
         // B_i and figure out we only have B_j.
         // Here we commit up to the highest_ledger_info to maintain highest_ledger_info == state_computer.committed_trees.
         if let Some(block_to_commit) = self.highest_ledger_info().committed_block_id() {
-            if block_to_commit != self.root().id() {
+            let highest_li_commit_round = self.get_block(block_to_commit).map_or(0, |b| b.round());
+            if block_to_commit != self.root().id() && highest_li_commit_round > self.root().round()
```

Comment:
Can you consider a case where block_0.id == block_1.id and block_0.round != block_1.round? Could this check be compressed to analyzing the round numbers only?

Reply:
Oh that's right. Checking the round is enough.